lib/src/http/common/request.dart

```diff
@@ -172,6 +172,15 @@ abstract class CommonRequest extends Object
         'The content-length of a request cannot be set manually when the request body is known in advance.');
   }
 
+  Duration get retryTimeoutThreshold {
```

Comment:
Maybe we should upper bound this to some reasonable value?

Reply:
I don't disagree, but given that increasing timeouts is opt-in and the base timeout is configurable, the user still has a decent amount of control. I'd vote for deferring this until we have a need for it.

Reply:
I think so.

Reply:
I just did it.